### PR TITLE
Fix several test subcommands.

### LINKED ISSUE #267: Game Test API: Doesn't work well with vanilla's test commands
Stuff I found:
- `TestFunctionArgumentType` uses `stringReader.readUnquotedString()`, which doesn't allow `:`, making it impossible to run tests using `/test run`
- `StringArgumentType.word()` does not accept colons, making `/test create` less useful
- `/test runall ClassName` doesn't seem to find the tests

### PR DESCRIPTION
**Pull request by LambdAurora**
Fix #267.

### CODE REVIEW

**Comment by LambdAurora** [comment]
Forgot the test class stuff, will fix tomorrow.

**Comment by LambdAurora** [comment]
The test class stuff has been fixed too, making this PR ready for review.

**Review by OroArmor — APPROVED** [review]

File: library/core/testing/src/main/java/org/quiltmc/qsl/testing/mixin/command/TestCommandMixin.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			method = "onCompletion",
+			constant = @Constant(stringValue = "All required tests passed :)")
+	)
+	private static String quiltGameTest$replaceSuccessMessage(String original) {
+		// You may ask why, it's simple.
+		// The original emoticon is a bit... weird.
+		// And QSL members expressed some kind of interest into replacing it.
+		// So here it is. I assure you this is a really necessary injection.
+		return "All required tests passed :3c";
+	}
```
Comment (EnnuiL) [inline library/core/testing/src/main/java/org/quiltmc/qsl/testing/mixin/command/TestCommandMixin.java:117]: I love this! /gen

**Review by EnnuiL — APPROVED** [review]
Lgtm!

File: library/core/testing/src/main/java/org/quiltmc/qsl/testing/mixin/command/TestCommandMixin.java (hunk 1 above)
Comment (Leo40Git) [reply library/core/testing/src/main/java/org/quiltmc/qsl/testing/mixin/command/TestCommandMixin.java:117]: oh my god you actually did it
based as all hell

**Pull request merged**